cilium/cmd/cleanup.go

```diff
@@ -511,3 +533,47 @@ func removeTCFilters(linkAndFilters map[string][]*netlink.BpfFilter) error {
 
 	return nil
 }
+
+func removeXDPs(links []netlink.Link) error {
+	for _, link := range links {
+		err := netlink.LinkSetXdpFd(link, -1)
+		if err != nil {
+			return err
+		}
+		log.Infof("removed cilium xdp of %s", link.Attrs().Name)
+	}
+	return nil
+}
+
+func isLinkAttachedCiliumXDP(link netlink.Link) (bool, error) {
+	linkxdp := link.Attrs().Xdp
+	if linkxdp == nil || !linkxdp.Attached {
+		return false, nil
+	}
+
+	ok, err := isCiliumXDP(linkxdp.ProgId)
+	if ok && err == nil {
+		return true, nil
+	}
+
+	return false, err
+}
+
+func isCiliumXDP(progId uint32) (bool, error) {
+	xdp, err := ebpf.NewProgramFromID(ebpf.ProgramID(progId))
+	if err != nil {
+		return false, err
+	}
+
+	info, err := xdp.Info()
+	if err != nil {
+		return false, err
+	}
+
+	if strings.Contains(info.Name, "bpf_xdp_entry") {
```

Comment:
How can we ensure that we won't forget to update this if/when we update the function's name in `bpf_xdp.c`? AFAIK, we don't have any tests covering this. At the very least, we should add a comment in `bpf_xdp.c`.